app/scripts/nav.js

```diff
@@ -27,18 +29,33 @@ $(function() {
 
     lastScrollTop = currentScrollTop;
 
-    if (isOnTop(currentScrollTop) && !nav.hasClass(onTopClass) ||
-        (!isOnTop(currentScrollTop) && nav.hasClass(onTopClass)))
-      toggleNavOnTop();
+    if (nav.hasClass(onTopClassIndex)) {
+      last_class = onTopClassIndex;
+    } else if (nav.hasClass(onTopClass)){
+      last_class = onTopClass;
+    }
+
+    if (last_class == onTopClass) {
```

Comment:
Please use the strictly equal operator: `===`. This way you make sure the two variables are equal in `type` and `value`.